research/02-repl.tex

```diff
@@ -1,5 +1,172 @@
 \section{Read-Eval-Print Loops}
-\label{sec:read-eval-print}
+\label{sec:repl}
+
+% TODO: mention that REPLs do not necessarily have to be interactive?
+Many programming languages come with an interactive environment. This
+interactive environment is an interface to the programming language's execution
+engine. These interactive environments know many different names, including but
+not limited to \textit{language shell}, \textit{command-line interpreter} and
+\textit{REPL}. There are subtle differences between these interpretations.
+These are, however, mostly of semantic value. In this report the term
```

Comment:
Perhaps I should say there are certain differences between the terms themselves, instead of their interpretations. The term REPL is open to two interpretations ([source](http://stackoverflow.com/questions/5671214/is-lisp-the-only-language-with-repl)):

1. A Lisp REPL is different to most other textual interactive interfaces, because it is based on the idea of data I/O of s-expressions and evaluating these.
2. A REPL is a general term describing textual interactive interfaces to programming language implementations or subsets of those.

Obviously we choose the second interpretation. Do you think I should add these two interpretations to the text?